### PR TITLE
Selective evidence flags for some items

### PR DESCRIPTION
**Pull request by Crosarius**
A chunky update to the forensic system. The core purpose of this PR is to make latex gloves, nitrile gloves, BST gloves & uniforms, and ninja equipment leave behind no evidence. No fibers, no fingerprints. 

However, I have also ended up patching up a few things here and there that I noticed were missing, such as a bug which caused opening an airlock by clicking it instead of bumping into it would not cause forensic data to be generated. 

Because this update makes nitrile gloves a direct upgrade to the black gloves that CSI and FT spawn with, they now spawn with them instead.

### CODE REVIEW

File: code/game/machinery/doors/airlock_control.dm [hunk 1]
```diff
@@ -263,7 +263,7 @@
 	..()
 
 /obj/machinery/access_button/attack_hand(mob/user)
-	add_fingerprint(usr)
+	src.add_fingerprint(usr)
```
Comment (mikomyazaki) [inline code/game/machinery/doors/airlock_control.dm:266]: This change does exactly nothing.

File: code/game/machinery/doors/door.dm [hunk 2]
```diff
@@ -270,6 +270,7 @@
 	return src.attack_hand(user)
 
 /obj/machinery/door/attack_hand(mob/user as mob)
+	src.add_fingerprint(user)
```
Comment (mikomyazaki) [inline code/game/machinery/doors/door.dm:273]: Ditto.

File: code/game/machinery/doors/firedoor.dm [hunk 3]
```diff
@@ -259,7 +259,7 @@
 
 /obj/machinery/door/firedoor/attackby(obj/item/C as obj, mob/user as mob)
 	if(!istype(C, /obj/item/forensics))
-		add_fingerprint(user)
+		src.add_fingerprint(user)
```
Comment (mikomyazaki) [inline code/game/machinery/doors/firedoor.dm:262]: Ditto.

File: code/game/machinery/doors/airlock_control.dm (hunk 1 above)
Comment (Crosarius) [reply code/game/machinery/doors/airlock_control.dm:266]: Trust me, after extensive testing I've found that for some fucked up reason it literally doesn't work at all if it doesn't use .src

File: code/game/machinery/doors/door.dm (hunk 2 above)
Comment (Crosarius) [reply code/game/machinery/doors/door.dm:273]: It seriously doesn't work if it's just add_fingerprint. Really, pull the master and just add that in as add_fingerprint and it won't work. It only works with .src - It's some fucky door thing or something idk.

File: code/modules/detectivework/forensics.dm [hunk 4]
```diff
@@ -11,8 +11,10 @@ proc/is_complete_print(var/print)
 atom/var/list/suit_fibers
 
 atom/proc/add_fibers(mob/living/carbon/human/M)
+	var/obj/item/clothing/gloves/S = M.wear_suit
+	var/obj/item/clothing/gloves/U = M.w_uniform
+	var/obj/item/clothing/gloves/G = M.gloves
```
Comment (NonQueueingMatt) [inline code/modules/detectivework/forensics.dm:16]: Only cast these when necessary, so when you know they're wearing it

File: code/game/machinery/doors/airlock_control.dm (hunk 1 above)
Comment (mikomyazaki) [reply code/game/machinery/doors/airlock_control.dm:266]: Then something else is horribly, horribly wrong.

File: code/modules/detectivework/forensics.dm (hunk 4 above)
Comment (Crosarius) [reply code/modules/detectivework/forensics.dm]: Fixed 👍

File: code/game/machinery/doors/airlock_control.dm (hunk 1 above)
Comment (Crosarius) [reply code/game/machinery/doors/airlock_control.dm:266]: I know right? Fingerprint code is fuuuucked but I don't know why. I just want to add my feature and GTFOutta it.

**Comment by Crosarius** [comment]
!review

**Pull request closed without merging**